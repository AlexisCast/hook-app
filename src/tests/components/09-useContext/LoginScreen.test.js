import React from "react";
import "@testing-library/jest-dom";
import { mount } from "enzyme";
import { UserContext } from "../../../components/09-useContext/UserContext";
import { LoginScreen } from "../../../components/09-useContext/LoginScreen";

describe("Prueba en <LoginScreen/>", () => {
	const setUser = jest.fn();
	const wrapper = mount(
		<UserContext.Provider
			value={{
				setUser,
			}}
		>
			<LoginScreen />
		</UserContext.Provider>
	);
	test("debe de mostrar correctamente", () => {
		expect(wrapper).toMatchSnapshot();
	});
	test("debe de ejecutar el seUser con el argumento esperado", () => {
		wrapper.find("button").prop("onClick")();
		expect(setUser).toHaveBeenCalledWith({
			id: 123,
			name: "Alexis",
		});
	});
});
