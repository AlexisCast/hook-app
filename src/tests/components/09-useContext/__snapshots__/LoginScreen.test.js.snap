// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <LoginScreen/> debe de mostrar correctamente 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
